src/common/config.cc

```diff
@@ -548,26 +548,26 @@ int md_config_t::parse_argv(std::vector<const char*>& args, int level)
        * argument parses will still need to see it. */
       break;
     }
-    else if (ceph_argparse_flag(args, i, "--show_conf", (char*)NULL)) {
+    else if (ceph_argparse_flag(args, i, "--show_conf", NULL)) {
```

Comment:
These casts are technically required because the size of pointers is permitted to differ based on type. (Be aware that all data pointers must be able to fit in a `void *` but the reverse is not necessarily true.)